src/gevent/_hub_primitives.py

```diff
@@ -101,30 +101,27 @@ def __init__(self, objects, hub, timeout, count):
         self._hub = hub
         self._waiter = MultipleWaiter(hub) # pylint:disable=undefined-variable
         self._switch = self._waiter.switch
-        self._timeout = timeout
+        self._timer = (
+            hub.loop.timer(timeout, priority=-1) if
+            timeout is not None else None
+        )
         self._objects = objects
 
-        self._timer = None
         self._begun = False
 
-        # Even if we're only going to return 1 object,
-        # we must still rawlink() *all* of them, so that no
-        # matter which one finishes first we find it.
-        self._count = len(objects) if count is None else min(count, len(objects))
+        self._objects = set(objects)
+        self._count = count or len(objects)
 
     def _begin(self):
         if self._begun:
             return
 
         self._begun = True
 
-        # XXX: If iteration doesn't actually happen, we
```

Comment:
This is still true, with the added proviso that the context manager is also not used. Please restore.